Sound/Tidal/Pattern.hs

```diff
@@ -1181,3 +1181,25 @@ swingBy::Time -> Time -> Pattern a -> Pattern a
 swingBy x n = inside n (within (0.5,1) (x ~>))
 
 swing = swingBy (1%3)
+
+{- | `cycleChoose` is like `choose` but only picks a new item from the list
+once each cycle -}
+cycleChoose::[a] -> Pattern a
+cycleChoose xs = Pattern $ \(s,e) -> [((s,e),(s,e), xs!!(floor $ (dlen xs)*(ctrand s) ))]
+  where dlen xs = fromIntegral $ length xs
+        ctrand s = timeToRand $ fromIntegral $ floor $ sam s
+
+{- | `shuffle n p` divides one cycle of the `p` into `n` parts and returns a
+new random permutation each cycle.  For example, `shuffle 4 "bd sn cp hh"` 
```

Comment:
ah so they are not the actual permutations, but random selections? So from a pattern `bd hh cp`` you could get the sequence `bd bd hh`, e.g. or are only `bd cp hh`, `cp bd hh` and `cp hh bd` possible?